lib/mina/local_helpers.rb

```diff
@@ -64,16 +61,18 @@ def invoke(script, this)
         code = "#{script}"
 
         # Certain environments can't do :pretty mode.
-        term_mode = :exec  if term_mode == :pretty && !pretty_supported?
+        term_mode = :exec if term_mode == :pretty && !pretty_supported?
 
         case term_mode
         when :pretty
+          require 'shellwords'
+          code = Shellwords.escape(code)
           this.pretty_system(code)
         when :exec
-          exec code
+          Kernel.exec code
         else
-          system code
-          $?.to_i
+          Kernel.system code
```

Comment:
changed to explicitly call exec/system on `Kernel` to 1) be able to test on the actual receiver and 2) be more clear about what's being done. several libraries often define their own `exec` or `system` methods (possibly delegating to `Kernel`) and my first impression was that this was happening here